File: omnes.Web/Modules/ServerTypes/Parametros/TiposEmailRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface TiposEmailRow {
    IdTipoEmail?: number;
    NombreTipoEmail?: string;
}

export abstract class TiposEmailRow {
    static readonly idProperty = 'IdTipoEmail';
    static readonly nameProperty = 'NombreTipoEmail';
    static readonly localTextPrefix = 'Parametros.TiposEmail';
    static readonly deletePermission = 'SuperAdmin';
    static readonly insertPermission = 'SuperAdmin';
    static readonly readPermission = 'SuperAdmin';
    static readonly updatePermission = 'SuperAdmin';

    static readonly Fields = fieldsProxy<TiposEmailRow>();
}

File: omnes.Web/Modules/Parametros/TiposEmail/TiposEmailRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace omnes.Parametros;

[ConnectionKey("Default"), Module("Parametros"), TableName("TiposEmail")]
[DisplayName("Tipos Email"), InstanceName("Tipos Email")]
[ReadPermission("SuperAdmin")]
[ModifyPermission("SuperAdmin")]
public sealed partial class TiposEmailRow : Row<TiposEmailRow.RowFields>, IIdRow, INameRow
{
    class RowTemplate
    {
        [DisplayName("Id Tipo Email"), Identity, IdProperty]
        public int? IdTipoEmail { get; set; }

        [DisplayName("Nombre Tipo Email"), Size(100), NotNull, QuickSearch, NameProperty]
        public string NombreTipoEmail { get; set; }
    }
}